src/api/mainPage.ts

```diff
@@ -0,0 +1,56 @@
+import { httpClient } from './index';
+
+interface RecommendItem {
+  itemId: number;
+  name: string;
+  brand: string;
+  price: number;
+  imgUrl: string;
+  like: boolean;
+}
+
+interface Recommendation {
+  bannerUrl: string | null;
+  title: string;
+  subtitle: string;
+  recommendItemList: RecommendItem[];
+}
+
+type RecommendationsResponse = Recommendation[];
+
+export interface GroupedRecommendations {
+  [title: string]: Recommendation[];
+}
+
+export interface GroupedRecommendationsResponse {
+  title: string;
+  recommendations: Recommendation[];
+}
+
+function groupByTitle(response: RecommendationsResponse) {
+  const keys: string[] = [];
+  const grouped: GroupedRecommendationsResponse[] = [];
+  let tempList: Recommendation[] = [];
+
+  response.forEach(recommendation => {
+    const { title } = recommendation;
+    tempList.push(recommendation);
+    if (!keys.includes(title)) {
+      keys.push(title);
+    } else {
+      grouped.push({
+        title,
+        recommendations: tempList,
+      });
+      tempList = [];
+    }
+  });
+
+  return grouped;
+}
```

Comment:
title이 같은 경우에는 묶으려고 하시는 것일까요? 만약 그렇다면 현재 로직에서 title이 같은 것이 3개라면 원하는 동작을 하지 않을 것 같습니다. 만약 title이 같은 경우가 2개로 고정이 아니라면 다음과 같이 수정하면 좋을 것 같습니다.

```suggestion
function groupByTitle(response: RecommendationsResponse) {
  const grouped: GroupedRecommendationsResponse[] = [];

  response.forEach(recommendation => {
    const { title } = recommendation;

    const index = grouped.findIndex(({ title: groupedTitle }) => groupedTitle === title);

    if (index === -1) {
      grouped.push({
        title,
        recommendations: [recommendation],
      });
    } else {
      grouped[index].recommendations.push(recommendation);
    }
  });

  return grouped;
}
```

Reply:
> title이 같은 경우에는 묶으려고 하시는 것일까요? 만약 그렇다면 현재 로직에서 title이 같은 것이 3개라면 원하는 동작을 하지 않을 것 같습니다. 만약 title이 같은 경우가 2개로 고정이 아니라면 다음과 같이 수정하면 좋을 것 같습니다.

제가 의도한것은 title이 여러개라도 묶는 것이긴합니다. 원래 API에서 묶어서 내려오면 좋을것 같은데 front에서 하는게 좋을것 같아서 했습니다. title로 보여주고 아래 subtitle 보여주려고 했어서 index 숫자 저장하면 JSX에서 보여줄 때 또 수정해야할것 같습니다 ㅠ @sky21kr

Reply:
> title이 같은 경우에는 묶으려고 하시는 것일까요? 만약 그렇다면 현재 로직에서 title이 같은 것이 3개라면 원하는 동작을 하지 않을 것 같습니다. 만약 title이 같은 경우가 2개로 고정이 아니라면 다음과 같이 수정하면 좋을 것 같습니다.

수정했습니다. 감사합니다. 테스트 코드 작성해보니 잘못된것을 바로 알 수 있었네요 @sky21kr